validator_test.go

```diff
@@ -301,14 +302,21 @@ func TestValidatorSubDomains(t *testing.T) {
 		},
 	}
 
-	dir, err := ioutil.TempDir("", "emailstest")
-	if err != nil {
-		log.Fatal(err)
-	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			g := NewWithT(t)
 
+			// Make a TempFile prefix somehow based on the testcase name
+			dir, err := ioutil.TempDir("", fmt.Sprintf("%s_", tc.name))
+			defer os.Remove(dir)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			defer func(t *testing.T) {
+				if err := os.RemoveAll(dir); err != nil {
+					t.Fatal(err)
+				}
+			}(t)
+
 			emailsFile := filepath.Join(dir, "emails")
 			if err := ioutil.WriteFile(emailsFile, []byte(strings.Join(tc.allowedEmails, "\n")), 0666); err != nil {
```

Comment:
This should be `0600` permissions otherwise Gosec security scans will complain.

Reply:
In general, what's the rationale for reinventing the wheel in this test case? The `ValidatorTest` struct in this package has helper methods to handle all the proper details of making temporary test files & cleanup (`WriteEmails` and `TearDown` methods).

If that test helper has any weaknesses, I think the proper route is to fix it appropriately so it can be used throughout the test suite still.

Reply:
Agree, I guess i misread the previous message. 
I've refactored the newly introduced tests to use the existing clean-up methods.